src/test/java/software/amazon/awssdk/crt/test/TlsContextOptionsTest.java

```diff
@@ -0,0 +1,46 @@
+package software.amazon.awssdk.crt.test;
+
+import static software.amazon.awssdk.crt.io.TlsContextOptions.TlsVersions;
+
+import org.junit.Assert;
+import org.junit.Test;
+import software.amazon.awssdk.crt.CrtResource;
+import software.amazon.awssdk.crt.io.TlsCipherPreference;
+import software.amazon.awssdk.crt.io.TlsContextOptions;
+
+public class TlsContextOptionsTest {
+
+    @Test
+    public void testTlsContextOptionsAPI() {
+        Assert.assertEquals(0, CrtResource.getAllocatedNativeResourceCount());
+
+        TlsContextOptions options = new TlsContextOptions();
+
+        for (TlsVersions tlsVersion: TlsContextOptions.TlsVersions.values()) {
+            options.setMinimumTlsVersion(tlsVersion);
+        }
+
+        options.setMinimumTlsVersion(TlsVersions.TLS_VER_SYS_DEFAULTS);
+
+        for (TlsCipherPreference pref: TlsCipherPreference.values()) {
+            if (TlsContextOptions.isCipherPreferenceSupported(pref)) {
+                options.setCipherPreference(pref);
+            }
+        }
+
+        boolean exceptionThrown = false;
```

Comment:
You could use junit's built-in [exception testing](https://github.com/junit-team/junit4/wiki/exception-testing)

Reply:
I wanted to catch the Exception so that I can call `options.close();` and not leak memory.

Reply:
That's fine, you might want to do a try with resources then.